fuzon-http/src/errors.rs

```diff
@@ -0,0 +1,7 @@
+#[derive(Debug, thiserror::Error, actix_web_error::Json)]
+#[status(BAD_REQUEST)] // default status for all variants
+pub enum ApiError {
```

Comment:
**praise:** very cool approach :100: 
**question**: iiuc, as we expand the API, we would add variants for the different failure cases to this enum?